doc/source/train/examples/lightning/lightning_exp_tracking.ipynb

```diff
@@ -131,9 +131,18 @@
     "\n",
     "    # Wandb\n",
     "    wandb_api_key = os.environ.get(\"WANDB_API_KEY\", None)\n",
-    "    wandb_logger = WandbLogger(\n",
+    "\n",
+    "    class RayWandbLogger(WandbLogger):\n",
+    "        # wandb.finish() ensures all artifacts get uploaded at the end of training.\n",
+    "        def finalize(self, status):\n",
+    "            super().finalize(status)\n",
+    "            wandb.finish()\n",
+    "\n",
```

Comment:
why don't we do this by default in the wandblogger?

Reply:
WandbLogger is a PyTorch lightning module. I'll create an issue to their repo.

More context here:
For performance concern, wandb asynchronously uploads artifacts and automatically triggers `wandb.finish()`(as a barrier) when your script exits, so PyTorch Lightning don't do this explicitly.

In Ray, we kill actors right after training is complete, so wandb may not be able to upload all files before the actor dies. So we have to explicitly call `finish()`.

Reply:
nice!